mat64/vector.go

```diff
@@ -71,11 +71,11 @@ func NewVector(n int, data []float64) *Vector {
 }
 
 // ViewVec returns a sub-vector view of the receiver starting at element i and
-// extending n columns. If i is out of range, or if n is zero or extend beyond the
-// bounds of the Vector ViewVec will panic with ErrIndexOutOfRange. The returned
-// Vector retains reference to the underlying vector.
+// extending n rows. If i is out of range, or if n is zero, or if the view
+// extends beyond the bounds of the Vector, ViewVec will panic with ErrIndexOutOfRange.
+// The returned Vector retains reference to the underlying vector.
 func (v *Vector) ViewVec(i, n int) *Vector {
-	if i+n > v.n {
+	if i < 0 || n <= 0 || i+n > v.n {
 		panic(matrix.ErrIndexOutOfRange)
 	}
 	return &Vector{
```

Comment:
I added a tight upper bound `(i+n-1)*v.mat.Inc+1` but now I realize that perhaps it may clash with some BLAS or other assumptions so I'd like to confirm that it is ok. If not, I'll change it to the simple `(i+n)*v.mat.Inc` and submit in a PR.